doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -267,6 +266,7 @@ There is dedicated syntax for list and array literals.
    | @ltac2_expr5
    ltac2_expr5 ::= fun {+ @tac2pat0 } => @ltac2_expr
    | let {? rec } @ltac2_let_clause {* with @ltac2_let_clause } in @ltac2_expr
+   | if @ltac2_expr5 then @ltac2_expr5 else @ltac2_expr5
```

Comment:
Shouldn't include this production here since it's styled as a tactic (as is `match`).

Reply:
I believe it's an error to group `match` and `match!` together. The two constructs have little in common, and the former is by no means a "tactic", whatever that means.

Reply:
(I'll write another PR after this one is merged.)

Reply:
We wanted Ltac and Ltac2 constructs such as `match` to be easy to find in the index.  Styling them as tactics includes them in the tactic index.  Syntactically they're generally accepted wherever tactics are accepted.  The general index is very long and not that easy to scan.  Nor could we justify creating yet another index just for these constructs.  Being a little imprecise on the notion of "tactic" seemed a reasonable compromise to make the doc more usable.  I think it would be pretty obvious to the reader that a `match` construct doesn't change the proof state aside from the tactics it contains.

> I believe it's an error to group `match` and `match!` together.

They're different subsections (see below).  Having them adjacent probably makes it easier for readers to notice that these are different but similarly-named constructs.

Despite spending considerable time refining the chapter, there's still lots more to improve here, including the overall structure and the presentation of Ltac2 concepts.

![image](https://user-images.githubusercontent.com/1253341/97905814-82175880-1cf7-11eb-84ca-707e9b4aa8ad.png)

Reply:
> Nor could we justify creating yet another index just for these constructs.

The `match` node is the basic ML pattern-matching, and as such should be part of the other core language syntax. Pattern-matching is as much primitive as function application and abstraction. Meanwhile, `match!` is syntactic sugar that expands to a call to a core primitive that deals with AST manipulation and backtracking, and is not part of the language but instead of the standard library. I don't see why `match` wouldn't go into the basic language description.

Reply:
OK, fair enough.  I suppose you mean to put it near and after the "Term Syntax" subsection.  Note the Ltac2 syntax is under-explained compared to the Ltac chapter (e.g. the latter explains `let` but Ltac2 doesn't explain its `let` or list it in the tactics index).